File: ls/server/src/main/java/org/opencds/cqf/cql/ls/server/repository/ig/standard/IgStandardResourceCategory.java
package org.opencds.cqf.cql.ls.server.repository.ig.standard;

import com.google.common.collect.Sets;
import java.util.Set;

enum IgStandardResourceCategory {
    DATA,
    TERMINOLOGY,
    CONTENT;

    private static final Set<String> TERMINOLOGY_RESOURCES = Sets.newHashSet("ValueSet", "CodeSystem");
    private static final Set<String> CONTENT_RESOURCES = Sets.newHashSet(
            "Library", "Questionnaire", "Measure", "PlanDefinition", "StructureDefinition", "ActivityDefinition");

    static IgStandardResourceCategory forType(String resourceType) {
        if (TERMINOLOGY_RESOURCES.contains(resourceType)) {
            return TERMINOLOGY;
        } else if (CONTENT_RESOURCES.contains(resourceType)) {
            return CONTENT;
        }

        return DATA;
    }
}

File: ls/server/src/main/kotlin/org/opencds/cqf/cql/ls/server/repository/ig/standard/IgStandardResourceCategory.kt
package org.opencds.cqf.cql.ls.server.repository.ig.standard

enum class IgStandardResourceCategory {
    DATA,
    TERMINOLOGY,
    CONTENT,
    ;

    companion object {
        private val TERMINOLOGY_RESOURCES = setOf("ValueSet", "CodeSystem")
        private val CONTENT_RESOURCES =
            setOf(
                "Library",
                "Questionnaire",
                "Measure",
                "PlanDefinition",
                "StructureDefinition",
                "ActivityDefinition",
            )

        fun forType(resourceType: String): IgStandardResourceCategory {
            return when {
                TERMINOLOGY_RESOURCES.contains(resourceType) -> TERMINOLOGY
                CONTENT_RESOURCES.contains(resourceType) -> CONTENT
                else -> DATA
            }
        }
    }
}
